src/solver/prelu/backward_prelu_multi_weights.cpp

```diff
@@ -0,0 +1,180 @@
+                                  const miopen::prelu::BackwardProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype        = problem.GetdInputDesc().GetType();
+    auto input_dtype  = miopen::GetDataType(problem.GetdInputDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetdOuputDesc().GetType());
+
+    /* Phase 1: Calc gradient for each elements. */
+    {
+        auto size         = problem.GetdInputDesc().GetElementSize();
+        auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_FP64", static_cast<int>(dtype == miopenDouble)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"VIEW_DIMS", VIEW_DIMS},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"OUTPUT_TYPE", output_dtype == "bfloat16" ? "ushort" : output_dtype},
+        };
+        result.construction_params.push_back(make_hip_kernel(
+            {LOCAL_SIZE_MW_BWD}, {size}, "MIOpenPReLU.cpp", "PReLUMWBackward", build_params));
+    }
+
+    /* Phase 2: Reduce gradient. */
+    {
+        auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_FP64", static_cast<int>(dtype == miopenDouble)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"OUTPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"REDUCE_SIZE", LOCAL_SIZE_MW_REDUCE_BWD},
+        };
+        result.construction_params.push_back(
+            make_hip_kernel({LOCAL_SIZE_MW_REDUCE_BWD},
+                            {problem.GetdWeightDesc().GetElementSize() * LOCAL_SIZE_MW_REDUCE_BWD},
+                            "MIOpenReduceSum.cpp",
+                            "Reduce1dSum",
+                            build_params));
+    }
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) params = raw_params.CastTo<miopen::prelu::InvokeParams>();
+
+            auto elapsed = 0.f;
+            HipEventPtr start;
+            HipEventPtr stop;
+
+            bool reset_profiling_state = false;
+            if(handle_.IsProfilingEnabled())
+            {
+                reset_profiling_state = true;
+                handle_.EnableProfiling(false);
+                hipStreamSynchronize(handle_.GetStream());
+                start = miopen::make_hip_event();
+                stop  = miopen::make_hip_event();
+                hipEventRecord(start.get(), handle_.GetStream());
+            }
```

Comment:
Just a little simplification (see the second part at the end of the function):
```suggestion
            bool profiling = handle_.IsProfilingEnabled();
            if(profiling)
            {
                handle_.EnableProfiling(false);
                hipStreamSynchronize(handle_.GetStream());
                start = miopen::make_hip_event();
                stop  = miopen::make_hip_event();
                hipEventRecord(start.get(), handle_.GetStream());
            }
```
Also it is applicable for the `singe_weight` version.

Reply:
Fixed it.